src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationProviderBase.cs

```diff
@@ -0,0 +1,446 @@
+
+        protected ISystemRuntimeInformation SystemRuntimeInformation => _systemRuntimeInformation;
+
+        public abstract int Priority { get; }
+
+        public abstract void PopulateWorkerConfigs(Dictionary<string, RpcWorkerConfig> configs);
+
+        internal static void AddProvider(WorkerConfigurationResolverOptions resolverOptions,
+                                    string workerName,
+                                    string workerDirPath,
+                                    IMetricsLogger metricsLogger,
+                                    IWorkerProfileManager profileManager,
+                                    ILogger logger,
+                                    ISystemRuntimeInformation systemRuntimeInformation,
+                                    Dictionary<string, RpcWorkerConfig> workerRuntimeToConfigMap)
+        {
+            using (metricsLogger.LatencyEvent(string.Format(MetricEventNames.AddProvider, workerDirPath)))
+            {
+                if (workerRuntimeToConfigMap.ContainsKey(workerName))
+                {
+                    return;
+                }
+
+                // After specialization, load worker config only for the specified runtime unless it's a multi-language app.
+                // Only skip worker directories that don't match the current runtime.
+                // Do not skip non-worker directories like the function app payload directory
+                if (ShouldSkipWorkerDirectory(resolverOptions.WorkerRuntime, Path.GetFileName(workerDirPath), resolverOptions.IsMultiLanguageWorkerEnvironment, resolverOptions.IsPlaceholderModeEnabled)
+                    && workerDirPath.StartsWith(resolverOptions.WorkersRootDirPath))
+                {
+                    return;
+                }
+
+                (var workerDescription, var workerConfigJson) = GetWorkerDescriptionAndConfig(workerDirPath, profileManager, resolverOptions.WorkerDescriptionOverrides, logger);
+                if (workerDescription is null || IsWorkerDescriptionDisabled(workerDescription, logger))
+                {
+                    return;
+                }
+
+                var workerConfig = BuildWorkerConfig(resolverOptions, workerDirPath, workerConfigJson, workerDescription, metricsLogger, logger, systemRuntimeInformation);
+                if (workerConfig is not null)
+                {
+                    workerRuntimeToConfigMap[workerName] = workerConfig;
+                }
+            }
+        }
+
+        internal static RpcWorkerConfig BuildWorkerConfig(WorkerConfigurationResolverOptions resolverOptions,
+                                            string workerDir,
+                                            JsonElement workerConfig,
+                                            RpcWorkerDescription workerDescription,
+                                            IMetricsLogger metricsLogger,
+                                            ILogger logger,
```

Comment:
This class already has some of these parameters as fields. Can this be an instance method and it accesses those?